experimental/calendar/src/calendar.rs

```diff
@@ -0,0 +1,47 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::{Date, DateDuration, DurationUnit, Iso};
+use std::fmt::Debug;
+
+/// A calendar implementation
+///
+/// Only implementors of [`Calendar`] should care about these methods, in general users of
+/// these calendars should use the methods on [`Date`] instead.
+///
+/// Individual [`Calendar`] implementations may have inherent utility methods
+/// allowing for direct construction, etc.
+pub trait Calendar {
+    type DateInner: PartialEq + Eq + Debug + Clone;
```

Comment:
Suggestion: If possible, remove the type bounds here and instead add them as bounds on the functions where you need them (like `impl PartialEq for Date`).

Reply:
Ping

Reply:
Oh, forgot to reply.

I consider these to be a part of the API of Date; it should be equality-comparible and cloneable (cloning in particular will be necessary for some APIs I think). I don't care as much for Debug but I figured we should be consistent.

Also otherwise it means that other users need to care about gnarly `<A::Calendar as Calendar>::DateInner: Debug` bounds, this way `Date` is always `Debug`

Reply:
Okay, that works for me for Eq, PartialEq, and Clone.  I have mixed feelings on Debug.  But I'm willing to revisit this once more code is written to see how often we would actually need the gnarly Debug bound.

Reply:
Action item: please track this action item somewhere (testing in a few months what happens when the Debug bound is removed).

Reply:
I'm okay with getting rid of Debug! It's the others I care about, I just added Debug for uniformity :smile: